pan3d/ui/analytics.py

```diff
@@ -0,0 +1,405 @@
+        self.state.figure_height = 50
+        self.state.show_group_by = False
+        self.state.show_temporal_slider = False
+        self.state.show_zonal_axis = True
+        self.state.show_update_button = False
+
+        with self:
+            with v3.VCardTitle():
+                with html.Div(classes="d-flex"):
+                    v3.VSelect(
+                        label="Plot Type",
+                        v_model=("active_plot", plot_options.get(PlotTypes.ZONAL)),
+                        items=("plots", list(plot_options.values())),
+                        hide_details=True,
+                        variant="outlined",
+                        density="compact",
+                    )
+                v3.VDivider(classes="my-2")
+                with html.Div(classes="d-flex"):
+                    v3.VSelect(
+                        v_show="show_zonal_axis",
+                        label="Zonal Axis",
+                        v_model=("zonal_axis", list(zonal_axes.keys())[0]),
+                        items=("axes", list(zonal_axes.keys())),
+                        hide_details=True,
+                        variant="outlined",
+                        density="compact",
+                    )
+                    v3.VSelect(
+                        v_show="show_group_by",
+                        label="Group By",
+                        v_model=("group_by", group_options.get(GroupBy.YEAR)),
+                        items=("groups", list(group_options.values())),
+                        hide_details=True,
+                        variant="outlined",
+                        density="compact",
+                    )
+                with html.Div(
+                    v_if="show_temporal_slider",
+                ):
+                    html.Div("Temporal Slice Selection")
+                    v3.VSlider(
+                        v_model=("temporal_slice", 0),
+                        min=0,
+                        max=("time_groups", 0),
+                        step=1,
+                        hide_details=True,
+                        density="compact",
+                    )
+                v3.VDivider(classes="my-2")
+                with html.Div(
+                    v_if="show_update_button", classes="text-center align-center"
+                ):
+                    v3.VBtn("Update Plots", click=self.update_large_plot)
+            with v3.VCardText(style=("`height: ${figure_height}%;`",)):
+                figure = plotly.Figure(
+                    display_logo=True,
+                    display_mode_bar=True,
+                )
+                self.ctrl.figure_update = figure.update
+
+    def expose_plot_specific_config(self):
+        """
+        Toggle visibility of components controlling plots based on chosen type
+        """
+        state = self.state
+        plot_type = state.active_plot
+        if plot_type == plot_options.get(PlotTypes.ZONAL):
+            state.figure_height = 50
```

Comment:
You could have some static setup in a map of PlotTypes so you could do something like

```python
config = PLOT_CONFIGS.get(plot_type, {})
state.update(config)
```